### PR TITLE
V1.2b enhance tests, add `Remark` and `DeleteCommand` related tests

### PR DESCRIPTION
**Pull request by UdhayaShan1**
## Description:
This pull request enhances the test coverage for the Realodex application by adding new tests and remarks related to the delete functionality. The changes aim to improve the robustness and reliability of the codebase by covering edge cases and potential future expansions.

## Changes Made:
-Added remark tests for future expansion.

-Implemented different remarks tests in EditPersonDescriptorTest and PersonTest.

-Added remarks to test subjects in CommandTestUtil.

-Included an optional remark test in AddCommandParserTest.

-Added some tests for DeleteCommandParser.

-Implemented a test for throwing a ParseException for invalid names in DeleteCommandParser.

-Added a small remark test to EditCommandParserTest.

-Included remark tests in ParserUtilTest.

-Added a delete by name parser test to RealodexParserTest.

## Purpose:
These changes aim to enhance the test suite for the Realodex application, ensuring better coverage and reliability of the codebase. By adding tests for delete functionality and including remarks for potential future expansions, we improve the overall quality of the software.

### CODE REVIEW

File: src/test/java/seedu/realodex/logic/parser/AddCommandParserTest.java [hunk 1]
```diff
@@ -239,6 +246,27 @@ public void parse_compulsoryFieldMissing_failure() {
                 expectedMessage);
     }
 
+    @Test
+    public void parse_optionalFieldMissing_success() {
```
Comment (UdhayaShan1) [inline src/test/java/seedu/realodex/logic/parser/AddCommandParserTest.java:250]: optional remark test

File: src/main/java/seedu/realodex/model/remark/Remark.java [hunk 2]
```diff
@@ -15,9 +17,21 @@ public class Remark {
      * @param remarkName A valid remark.
      */
     public Remark(String remarkName) {
+        requireNonNull(remarkName);
         this.remarkName = remarkName;
     }
 
+    /**
+     * Returns true if a given string is a valid name.
+     * However, as of v1.2 there is no concrete check; hence it will always return true.
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/model/remark/Remark.java:26]: is this even a valid javadoc

Comment (jyztintan) [reply src/main/java/seedu/realodex/model/remark/Remark.java:26]: 🤔

File: src/main/java/seedu/realodex/model/remark/Remark.java [hunk 3]
```diff
@@ -15,9 +17,21 @@ public class Remark {
...
+     * @param test
+     * @return
+     */
```
Comment (jyztintan) [inline src/main/java/seedu/realodex/model/remark/Remark.java:29]:
```suggestion
 /**
 * Validates if the given string is a valid remark. In the current implementation (as of v1.2),
 * this method does not perform any actual validation checks and will always return true.
 * This is a placeholder implementation and may change in future versions.
 *
 * @param test the string to be validated as a remark. Cannot be null.
 * @return always true in the current version.
 * @throws NullPointerException if the test parameter is null.
 */
```

credits to chatgpt 4.0

File: src/main/java/seedu/realodex/model/remark/Remark.java (hunk 2 above)
Comment (jyztintan) [reply src/main/java/seedu/realodex/model/remark/Remark.java:26]: actually what would be a "invalid remark"

File: src/test/java/seedu/realodex/logic/commands/EditPersonDescriptorTest.java [hunk 4]
```diff
@@ -55,6 +56,10 @@ public void equals() {
         // different tags -> returns false
         editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withTags(VALID_TAG_AMY).build();
         assertFalse(DESC_AMY.equals(editedAmy));
+
+        //different remarks -> returns false
+        editedAmy = new EditPersonDescriptorBuilder(DESC_AMY).withRemark(VALID_REMARK_BOB).build();
+        assertFalse(DESC_AMY.equals(editedAmy));
```
Comment (jyztintan) [inline src/test/java/seedu/realodex/logic/commands/EditPersonDescriptorTest.java:62]: Good check

File: src/test/java/seedu/realodex/logic/parser/RealodexParserTest.java
Comment (jyztintan) [inline src/test/java/seedu/realodex/logic/parser/RealodexParserTest.java:1]: These checks are good to test our new implementations, as we can now delete by index or delete by name

File: src/test/java/seedu/realodex/model/person/RemarkTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        // null name
+        assertThrows(NullPointerException.class, () -> Remark.isValidRemark(null));
+
+        // invalid name
+        // to-do in the future, ideas include but not limited to length constraints, no illegal characters.
+
+        // valid name
+        assertTrue(Remark.isValidRemark("I love apples")); // alphabets only
+        assertTrue(Remark.isValidRemark("12345")); // numbers only
+        assertTrue(Remark.isValidRemark("James Lau the 1st")); // alphanumeric characters
```
Comment (jyztintan) [inline src/test/java/seedu/realodex/model/person/RemarkTest.java:30]: 🫡🫡🫡

**Review by jyztintan — APPROVED** [review]
LGTM

File: src/test/java/seedu/realodex/logic/commands/EditPersonDescriptorTest.java (hunk 4 above)
Comment (UdhayaShan1) [reply src/test/java/seedu/realodex/logic/commands/EditPersonDescriptorTest.java:62]: Not sure, maybe too long remark or something, I worried they may input something too long and UI may overflow or smth

File: src/test/java/seedu/realodex/logic/parser/RealodexParserTest.java
Comment (UdhayaShan1) [reply src/test/java/seedu/realodex/logic/parser/RealodexParserTest.java:1]: Thanks!
